pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -161,6 +161,11 @@ func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object, k8s ku
 		return obj, nil
 	case *v1.Pipeline:
 		// TODO(#6356): Support V1 Task verification
+		// Validation of beta fields must happen before the V1 Pipeline is converted into the storage version of the API.
+		// TODO(#6592): Decouple API versioning from feature versioning
+		if err := obj.Spec.ValidateBetaFields(ctx); err != nil {
+			return nil, fmt.Errorf("invalid remote Pipeline %s: %w", obj.GetName(), err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("invalid Pipeline %s: %w", obj.GetName(), err)
```

I don't we use the term "remote pipeline" consistently so it may be a bit confusing for the user

Reply:
updated-- we were also using it for the trusted resources error so I removed that as well.